### ISSUE TITLE
Rule F403 behaves weirdly

### ISSUE THREAD

**Issue by Saransh-cpp** [astral-sh/ruff#7176]
Taking the example from ruff's [docs](https://beta.ruff.rs/docs/rules/undefined-local-with-import-star/) -

If I run ruff over the following code snippet -

```py
from math import *


def area(radius):
    return pi * radius**2
```

it errors out with -
```
F405 `pi` may be undefined, or defined from star imports
```

Shouldn't this error out with rule F403?

Moreover, I want ruff to not error out at all  if I do the following -
```py
from math import *  # noqa: F403


def area(radius):
    return pi * radius**2
```

but ruff instead errors out with the following -
```
RUF100 [*] Unused `noqa` directive (unused: `F403`)
F405 `pi` may be undefined, or defined from star imports
```

---

command used:
`pre-commit run ruff --all-files`

pre-commit config:
```yaml
- repo: https://github.com/astral-sh/ruff-pre-commit
    rev: "v0.0.282"
    hooks:
      - id: ruff
        args: []
```
Am I missing something here? Thanks!

**Comment by Saransh-cpp** [comment]
False alarm, this was an issue with my ruff config. Sorry!
